pkg/apis/logicalcluster/v1alpha1/logical-cluster.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LogicalCluster represents a cluster.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster,shortName=ecl
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type LogicalCluster struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// `spec` describes a cluster.
+	// +optional
+	Spec LogicalClusterSpec `json:"spec,omitempty"`
+
+	// `status` describes the status of the cluster object.
+	// +optional
+	Status LogicalClusterStatus `json:"status"`
+}
+
+// LogicalClusterSpec describes a cluster.
+type LogicalClusterSpec struct {
+	// ClusterProviderDesc is a reference to a ClusterProviderDesc resource
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ProviderName is immutable"
+	ClusterProviderDesc string `json:"ClusterProviderDesc"`
+
+	// ClusterName is the name of the logical cluster this workspace is stored under.
```

Comment:
What, here, is a "workspace"?
How is the name here different from the `ObjectMeta.Name` ?

Reply:
typo. This is the logical cluster name. Will fix the comment

Reply:
Fixed. https://github.com/kcp-dev/edge-mc/pull/607/commits/febc2d00536c20cf80e0c173a586f26d5bec0f18